workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -287,24 +288,74 @@ describe('DynamicAuthorizationService', () => {
   });
 
   describe('addUserToGroup', () => {
-    it('returns userID and groupID when user was successfully added to the group', async () => {
-      mockGroupManagementPlugin.addUserToGroup = jest.fn().mockResolvedValue({ data: { userId, groupId } });
+    let auditServiceWriteSpy: jest.SpyInstance;
+    let actor: object;
+    let source: object;
+    let action: string;
 
-      const { data } = await dynamicAuthzService.addUserToGroup({
-        groupId,
-        userId,
+    beforeAll(() => {
+      auditServiceWriteSpy = jest.spyOn(auditService, 'write');
+
+      action = 'addUserToGroup';
+      actor = mockUser;
+      source = {
+        serviceName: 'DynamicAuthorizationService'
+      };
+    });
+
+    afterEach(jest.resetAllMocks);
+
+    it('returns userID and groupID when user was successfully added to group', async () => {
+      const mockReturnValue = { data: { userId: 'userId', groupId: 'groupId' } };
+      mockGroupManagementPlugin.addUserToGroup = jest.fn().mockResolvedValue(mockReturnValue);
+
+      const requestBody = {
+        groupId: 'groupId',
+        userId: 'userId',
         authenticatedUser: mockUser
-      });
+      };
+
+      const { data } = await dynamicAuthzService.addUserToGroup(requestBody);
+
+      expect(mockGroupManagementPlugin.addUserToGroup).toBeCalledWith(requestBody);
+      expect(data).toStrictEqual(mockReturnValue.data);
 
-      expect(data).toStrictEqual({ userId, groupId });
+      expect(auditServiceWriteSpy).toHaveBeenCalledWith(
+        {
+          actor,
+          source,
+          action,
+          requestBody,
+          statusCode: 200
+        },
+        mockReturnValue
+      );
     });
 
-    it('throws when the user cannot be added', async () => {
-      mockGroupManagementPlugin.addUserToGroup = jest.fn().mockRejectedValue(new GroupNotFoundError());
+    it('throws and writes to audit service when user cannnot be added to group', async () => {
+      mockGroupManagementPlugin.addUserToGroup = jest
+        .fn()
+        .mockRejectedValue(new RoleNotFoundError('Role does not exist.'));
```

Comment:
This should throw a `GroupNotFoundError`

Reply:
Done.